pinot-core/src/test/java/org/apache/pinot/queries/H3IndexQueriesTest.java

```diff
@@ -210,11 +226,47 @@ public void testH3Index()
       Assert.assertNotNull(aggregationResult);
       Assert.assertEquals((long) aggregationResult.get(0), NUM_RECORDS);
     }
+
+    // Test st contains in polygon
+    testQueryStContain("SELECT COUNT(*) FROM testTable WHERE ST_Contains(ST_GeomFromText('POLYGON ((\n"
```

Comment:
shall we add some tests of the corner cases, such as on the line/vertex of the polygon?

Reply:
sure.

Reply:
The ST_Contains seems doesn't include the point or line lying in polygon boundary. https://postgis.net/docs/ST_Contains.html
but let me try create a test with points very close to the boundary.